compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -626,9 +626,6 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
 
         let item = f(self);
         debug_assert_eq!(def_id, item.def_id());
-        // `f` should have consumed all the elements in these vectors when constructing `item`.
-        debug_assert!(self.impl_trait_defs.is_empty());
-        debug_assert!(self.impl_trait_bounds.is_empty());
```

Comment:
Why aren't those consumed any more?  This assertions is here to ensure we do not forget nodes during lowering.

Reply:
Right, this shouldn't have been removed. It was part of a hacky way I've used to fix a problem and then after doing the right thing, I should have introduced it back. Fixed https://github.com/rust-lang/rust/pull/101345/commits/0fadeede0b5631fd13eb78fcd38e7ae1608e5f44